### PR TITLE
Document scroll param on reindex.json

### PR DESCRIPTION
**Pull request by Mpdreamz**
The Reindex API also exposes `scroll` as a querystring parameter.

cc @elastic/es-clients

### CODE REVIEW

**Review by javanna — APPROVED** [review]
LGTM

File: rest-api-spec/src/main/resources/rest-api-spec/api/reindex.json [hunk 1]
```diff
@@ -30,6 +30,11 @@
           "default": 0,
           "description": "The throttle to set on this request in sub-requests per second. -1 means no throttle."
         },
+        "scroll": {
+          "type" : "time",
+          "description" : "Control how long to keep the search context alive",
+          "default" : "5m"
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/api/reindex.json]: where did you find the 5m default again?

**Comment by javanna** [comment]
this seems to be a duplicate of #31316, but the content is slightly different as this one contains the default. @Mpdreamz can you clarify that so we decide which one of the two PRs we should get in? :)

File: rest-api-spec/src/main/resources/rest-api-spec/api/reindex.json (hunk 1 above)
Comment (Mpdreamz) [reply rest-api-spec/src/main/resources/rest-api-spec/api/reindex.json]: > Both work exactly how they work in the Bulk API. As _reindex uses scroll search, you can also specify the scroll parameter to control how long it keeps the "search context" alive, (e.g. ?scroll=10m). The default value is 5 minutes.

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html

**Comment by Mpdreamz** [comment]
run elasticsearch-ci/bwc

**Comment by jpountz** [comment]
@Mpdreamz  I'm assuming there is nothing left to backport and removed the backport pending label.

**Pull request merged**